dynparquet/schema.go

```diff
@@ -431,6 +445,31 @@ func (s *Schema) NewWriter(w io.Writer, dynamicColumns map[string][]string) (*pa
 	), nil
 }
 
+type PooledWriter struct {
+	key string
+	*parquet.Writer
+}
+
+func (s *Schema) GetWriter(w io.Writer, dynamicColumns map[string][]string) (*PooledWriter, error) {
+	key := serializeDynamicColumns(dynamicColumns)
```

Comment:
Technically I need to also sort each suffix too at the risk of the cache not working, now sure if that's required or not that could be a big to do each time.

Reply:
Not sure exactly if this is what you mean but keys are sorted, and values for keys are sorted, so there _should_ be no different keys with the same dynamic columns.